specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json

```diff
@@ -2031,6 +2110,32 @@
         }
       }
     },
+	"CheckNameAvailabilityResult": {
```

Comment:
This should align with the following:

{
  "nameAvailable": true|false,
  "reason": "Invalid|AlreadyExists",
  "message": "<error message>"
}

https://github.com/Azure/azure-resource-manager-rpc/blob/002d272ec9c522f1b8606c5b062a9fa80145efe8/v1.0/proxy-api-reference.md#check-name-availability-requests

Is this already GA'd in this form?

Reply:
Yes, it's already in production and unfortunately this might be a breaking change. Can we please keep it in that way?

The reason we used enum instead of boolean for "nameAvailable" is because of the linter warning: 

WARNING (EnumInsteadOfBoolean/R3018/ARMViolation): Booleans are not descriptive and make them hard to use. Consider using string enums with allowed set of values defined. 

What should be the correct behavior in that case?

Reply:
I feel the same way about the bool here but its the pattern for checkNameAvailability. If this is already out in a GA'd api version then we can move forward with just documenting the behavior, even if its not perfectly aligned. Consider aligning in a future api version though.

Reply:
It's already GA'd, so we'll note it and keep that in mind for the next api version